README.md

```diff
@@ -3,13 +3,37 @@
 - List 3 reasons why asymptotic analysis may be misleading with respect to
   actual performance in practice.
 
+  - asymptotic analysis is used to compare growth rates for algorithms. This means that constant factors and lower order factors are ignored. a function that has a practical runtime of 3n^2 would have a better runtime on smaller data sets than a function with a runtime of 50n, even though an asymptotic analysis would result in them being O(n^2) and O(n) respectively. For example if we know that the data set we're using is only going to consist of a first middle and last initial of people, so three elements for each run, our O(n^2) runtime would actually come out to 27, while our O(n) runtime would come out to 150. 
+    
+  - asymptotic analysis operates under assumptions about the data set used, ie best case, worst case, average case. In reality data used may adhere to certain patterns, such as always being partially or nearly sorted. algorithms that have good best case but bad average or worst case complexity may be faster on said data sets, while an algorithm with the same best worst and average case may be less effective. Asymptotic analysis is a tool to quantify and compare algorithms based on assumptions that in reality may not be true. Asymptotic complexity gives a heirarchy of speeds, constant time faster than logarithmic time faster than linear time etc. but this heirarchy does not alwady remain consistent when used in real world applications, on data sets like in the examples given. More has to be known about the problem and the algorithm than asymptotic complexity of a given algorithm to determine the fastest option. For example if we did an asymptotic analysis of insertion sort and heapsort, a person might be immediately dissuaded by insertions much worse average case time complexity when compared to heapsort. However if we know that our data is going to be completely or nearly sorted every time, insertion sort would actually save time over heapsort, due to how it interacts with that specific pattern of data.
```

Comment:
I don't see how the analysis is misleading if you're incorrectly assuming the worst case.

Reply:
real world data may have underlying patterns that a person would not inherently know when choosing an algorithm. Because of this they maybe opt to choose the algorithm with the best average time complexity. But as illustrated in my example, the underlying patterns of certain data may make different algorithms preferable under certain circumstances that are not accounted for by asymptotic analysis

Reply:
But all of this would mean that the analysis was done incorrectly.

Reply:
so would my entire point be void because the idea of real world data not adhering to the conclusions drawn by asymptotic analysis is an analysis done incorrectly?

Reply:
Not necessarily, but I'm looking for something more concrete.